File: library/src/main/java/androidz/Androidz.java
package androidz;

import android.app.Application;
import android.content.Context;
import android.os.Handler;
import android.os.Looper;

/**
 * Main
 */
public class Androidz {
    private static final Handler handler = new Handler(Looper.getMainLooper());
    private static boolean debuggable;

    private Androidz() {
    }

    public static void initialize(Context context) {
        App.attachApplication((Application) context.getApplicationContext());
        debuggable = isDebuggable();
        if (isDebuggable()) {
            ActivityStackManager.getInstance().register((Application) context);
        }
    }

    public static Application getApp() {
        return App.INSTANCE;
    }

    public static boolean isDebuggable() {
        return debuggable;
    }

    public static void setDebuggable(boolean debuggable) {
        Androidz.debuggable = debuggable;
    }

    public static Handler getHandler() {
        return handler;
    }

    public static boolean isMainThread() {
        return Looper.getMainLooper().getThread() == Thread.currentThread();
    }

    public static void setLogger(Logger logger) {
        Logger.Companion.setDefault(logger);
    }
}

File: library/src/main/java/androidz/Androidz.kt
package androidz

import android.app.Application
import android.content.Context
import android.os.Handler
import android.os.Looper
import android.util.Log

/**
 * Main
 */
object Androidz {

    @JvmStatic
    fun initialize(context: Context) {
        App.attachApplication(context.applicationContext as Application)
        isDebuggable = App.isDebuggable
        ActivityStackManager.register(app)
        ActivityStackManager.isDebuggable = isDebuggable
    }

    @JvmStatic
    val app: App = App

    @JvmStatic
    val handler: Handler = Handler(Looper.getMainLooper())

    @JvmStatic
    var isDebuggable: Boolean = false

    @JvmStatic
    var logger: Logger = object : Logger() {
        override fun log(priority: Int, tag: String, msg: String, tr: Throwable?) {
            if (!isDebuggable)
                return
            when (priority) {
                Log.VERBOSE -> Log.v(tag, msg, tr)
                Log.DEBUG -> Log.d(tag, msg, tr)
                Log.INFO -> Log.i(tag, msg, tr)
                Log.WARN -> Log.w(tag, msg, tr)
                Log.ERROR -> Log.e(tag, msg, tr)
                else -> Log.v(tag, msg, tr)
            }
        }
    }

    @JvmStatic
    val isMainThread: Boolean
        get() = Looper.getMainLooper().thread === Thread.currentThread()

    @JvmStatic
    val isBackgroundThread: Boolean
        get() = !isMainThread

    @JvmStatic
    @JvmOverloads
    fun runOnUiThread(r: Runnable, delayMillis: Long = 0) {
        handler.postDelayed(r, delayMillis)
    }

}
